src/dune_rules/ocaml_toolchain.mli

```diff
@@ -26,6 +26,12 @@ val of_env_with_findlib
 
 val of_binaries : path:Path.t list -> Context_name.t -> Env.t -> Path.Set.t -> t Memo.t
 
+val of_toolchain_compiler
```

Comment:
Add a note that this is where it will attempt to download and build the compiler (it it doesn't exist).

Reply:
I assume this comment was left before introducing a witness type into `Toolchain.Compiler`, as now `of_toolchain_compiler` no longer installs the compiler (it takes a compiler that's already been installed).

Reply:
Yep, I addressed some of the notes of the review we did with @moyodiallo and @maiste but not all of them.

Reply:
I moved this entire function to where we setup the toolchain modules in Pkg_rules. Now all the implementation details for the rules live in one place (more or less)